packages/blocks/src/lib/slam-map-2d/motion-path.svelte

```diff
@@ -0,0 +1,57 @@
+<!--
+  @component
+  Renders a motion plan using thick lines.
+  Assumes the motion plan is coming in as a string in the format:
+  x,y\n
+  x,y\n
+  ...
+-->
+<script lang='ts'>
+
+import { T, extend } from '@threlte/core';
+import { Line2 } from 'three/examples/jsm/lines/Line2.js';
+import { LineMaterial } from 'three/examples/jsm/lines/LineMaterial.js';
+import { LineGeometry } from 'three/examples/jsm/lines/LineGeometry.js';
+import { renderOrder } from './constants';
+
+extend({ Line2, LineMaterial });
+
+export let path: string | undefined;
+
+let geometry: LineGeometry = new LineGeometry();
+
+const updatePath = (pathstr?: string) => {
```

Comment:
Should this function be extracted and unit tested?

Reply:
Yes! Just not yet - this is an advanced debugging feature that really only has internal use. There's a project in the works to hook the feature up to an API, at which point the incoming path data and therefore the function body is very likely to change. We should do it then if necessary (might just end up being plugging the response into the renderer rather than any parsing).